sdk/core/Azure.Core/src/Internal/ArrayBackedPropertyBag.cs

```diff
@@ -16,31 +14,17 @@ namespace Azure.Core
     /// </summary>
     internal struct ArrayBackedPropertyBag<TKey, TValue> where TKey : struct, IEquatable<TKey>
     {
-        private Kvp _first;
-        private Kvp _second;
-        private Kvp[]? _rest;
+        private (TKey Key, TValue Value) _first;
+        private (TKey Key, TValue Value) _second;
+        private (TKey Key, TValue Value)[]? _rest;
         private int _count;
 #if DEBUG
         private bool _disposed;
 #endif
-        private readonly struct Kvp
-        {
-            public readonly TKey Key;
-            public readonly TValue Value;
-
-            public Kvp(TKey key, TValue value)
-            {
-                Key = key;
-                Value = value;
-            }
-
-            public void Deconstruct(out TKey key, out TValue value)
-            {
-                key = Key;
-                value = Value;
-            }
+        private readonly object _lock = new();
```

Comment:
Why do you need this lock?

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/41085#discussion_r1447683483